src/schoology-extractor/schoology_extractor/api/request_client.py

```diff
@@ -74,61 +102,167 @@ def _get(self, url):
         )
         return response.json()
```

Comment:
What happens if `requests-oauthlib` encounters an HTTP response code of 4xx or 5xx? I recommend that you do some experimentation to see if you can trigger 4xx response. Triggering a 5xx response would be more difficult. See how this library and the main request library handle non-2xx error codes. Whatever they do for 4xx may be similar to 5xx - but we might need to build a quick-and-dirty web service that returns 500 responses so that we can test again that and make sure we understand the response. I want to make sure have good messages for the user. We also need to think about retries at that point.

I'm going to create a separate ticket for this point. I don't want to delay this ticket with this additional scope, and I don't think this scope should go into the logging ticket. No need to make any changes right now.

Reply:
There's already a ticket for that 😆 . FIZZ-50.

Reply:
It fails in this line when response is empty(not 200)... I think we have a ticket for error handling: https://tracker.ed-fi.org/browse/FIZZ-50